File: 1/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        const size_t arrayLen = nums.size();
        map<int, int> complementElement;
        
        for (int idx=0; idx < arrayLen; idx++) {
            int complementNum = target - nums[idx];
            if (complementElement.find(complementNum) != complementElement.end()) {
                map<int, int>::iterator ele = complementElement.find(complementNum);
                return {ele->second, idx};
            }
            complementElement[nums[idx]] = idx;
        }
        return vector<int>();
    }
};

File: 1/solution.go
func twoSum(nums []int, target int) []int {
	complacencyMap := make(map[int]int)
	for idx, val := range nums {
		compleNum := target - val
		frontIdx, ok := complacencyMap[compleNum]
		if ok {
			return []int{frontIdx, idx}
		}
		complacencyMap[val] = idx
	}
	return []int{0, 0}
}